libr/bin/bin.c

```diff
@@ -246,7 +246,7 @@ static void get_strings_range(RBinFile *arch, RList *list, int min, ut64 from, u
 	if (min < 0)
 		return;
 
-	if (!to || to > arch->buf->length)
+	if (!to || to > arch->size)//arch->buf->length)
```

Comment:
this is causing oob read because after that in `string_scan_range`

```
...
 rc = r_utf8_decode (buf + needle, to - needle, NULL);
...
```

here `to` goes beyond buf and crashing the app (we lose the buffer's boundaries).

```
mac:elf alvaro_fe$ export R_IO_MAX_ALLOC=4096
mac:elf alvaro_fe$ r2 /bin/ls
WARNING: File is greater than 0x1000 bytes.
Allocating R_IO_MAX_ALLOC set as the environment variable.
error parsing symtab
Warning: Cannot initialize items
WARNING: bin_strings buffer is too big (0xffffffffffffc700). Use -zzz or set bin.maxstrbuf (RABIN2_MAXSTRBUF) in r2 (rabin2)
WARNING: bin_strings buffer is too big (0xffffffffffffc510). Use -zzz or set bin.maxstrbuf (RABIN2_MAXSTRBUF) in r2 (rabin2)
WARNING: bin_strings buffer is too big (0xffffffffffffbd60). Use -zzz or set bin.maxstrbuf (RABIN2_MAXSTRBUF) in r2 (rabin2)
 -- Thank you for using radare2. Have a nice night!
[0x100000e94]> is
[Symbols]

0 symbols

[0x100000e94]>
```
bin->size in case of mach is set to buf->length, change it to r_io_size since we are checking along the code against this value to avoid oob read/writes
```
struct MACH0_(obj_t)* MACH0_(new_buf)(RBuffer *buf) {
	struct MACH0_(obj_t) *bin = R_NEW0 (struct MACH0_(obj_t));
	if (!bin) return NULL;
	bin->kv = sdb_new (NULL, "bin.mach0", 0);
	bin->b = r_buf_new ();
	bin->size = buf->length;
	if (!r_buf_set_bytes (bin->b, buf->buf, bin->size)){
		return MACH0_(mach0_free) (bin);
	}
	if (!init(bin))
		return MACH0_(mach0_free)(bin);
	return bin;

```